src/Domain/CoordinateMaps/Frustum.cpp

```diff
@@ -241,12 +259,16 @@ std::optional<std::array<double, 3>> Frustum::inverse(
         coord /= abs(coord);
       }
     }
-    logical_coords = RootFinder::gsl_multiroot(
-        rootfunction, logical_coords, absolute_tolerance, maximum_iterations,
-        relative_tolerance, verbosity, max_absolute_tolerance, method,
-        condition);
+    try {
+      logical_coords = RootFinder::gsl_multiroot(
+          rootfunction, logical_coords, absolute_tolerance, maximum_iterations,
+          relative_tolerance, verbosity, max_absolute_tolerance, method,
+          condition);
+
+    } catch (const convergence_error& e) {
+      return std::nullopt;
```

Comment:
Can you add a tests that gives us confidence that the inversion works correctly? Perhaps create an equiangular frustum and probe a bunch of points inside, outside, and right on the boundary. I'm thinking that inverse(map(xi)) should always work for points inside and on the boundary, and inverse(x) should always return nullopt for points outside. 

Are there any points that lie inside the frustum if it is equidistant, but lie slightly outside if it is equiangular? I thought this was the cause for Niko's issue, but now I am unsure because the frustum doesn't actually change shape right? So I still don't quite understand the failure mode that Niko encountered.

Reply:
Hi Nils, this should be covered ny the test suite for maps on the unit cube (At least, the testing of multiple points inside and on the boundaries of the Frustum). The test_frustum_fail tests that points on the outside of the Frustum, but do we want to return nullopt for every and all points outside [-1,1]^3? Because sometimes the point is invertible even when the point is outside, and the try catch block should cover the cases where it isn't. Is there a case you have in mind that isn't covered here?

As for equiangular vs equidistant, the equiangular map maps [-1,1] to itself, so points inside [-1,1]^3 should be inside in both cases, up to roundoff error.